R/data.R

```diff
@@ -204,27 +204,77 @@ any_zero_dims <- function(data) {
   any(has_zero_dims)
 }
 
-#' Write posterior draws objects to csv files
-#' @noRd
-#' @param draws A `draws_array` from posterior pkg
-#' @param sampler_diagnostics Either `NULL` or a `draws_array` of sampler diagnostics
+#' Write posterior draws objects to CSV files suitable for running standalone generated
+#' quantities with CmdStan.
+#'
+#' @export
+#' @param draws A `posterior::draws_*` object.
+#' @param sampler_diagnostics Either `NULL` or a `posterior::draws_*` object
+#'  of sampler diagnostics.
+#' @param dir (string) An optional path to the directory where the CSV files will be
+#'   written. If not set, [temporary directory][base::tempdir] is used.
+#' @param basename (string) If `dir` is specified, `basename`` is used for naming
+#' the output CSV files. If not specified, the file names are randomly generated.
+#'
 #' @return Paths to CSV files (one per chain).
 #'
-draws_to_csv <- function(draws, sampler_diagnostics = NULL) {
+#' @details
+#' `draws_to_csv()` generates a CSV suitable for running standalone generated
+#' quantities with CmdStan. The CSV file contains a single comment `#num_samples`,
+#' which equals the number of iterations in the supplied draws object.
+#'
+#' The comment is followed by the column names. The first column is the `lp__` value,
+#' followed by sampler diagnostics and finnaly other variables of the draws object.
+#' #' If the draws object does not contain the `lp__` or sampler diagnostics variables,
+#' columns with zeros are created in order to conform with the requirements of the
+#' standalone generated quantities method of CmdStan.
+#'
+#' The column names line is finally followed by the values of the draws in the same
+#' order as the column names.
+#'
+#' @examples
+#' \dontrun{
+#' draws <- posterior::example_draws()
+#'
+#' draws_csv_files <- draws_to_csv(draws)
+#' print(draws_csv_files)
+#'
+#' # draws_csv_files <- draws_to_csv(draws,
+#' #                                 sampler_diagnostic = sampler_diagnostics,
+#' #                                 dir = "~/my_folder",
+#' #                                 basename = "my-samples")
```

Comment:
I commented this out because when we generate the website we have it run all examples, even the onces wrapped in `\dontrun{}` (the `\dontrun{}` will just be for CRAN).